sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/StorageApiWritesShardedRecords.java

```diff
@@ -247,7 +251,13 @@ public PCollectionTuple expand(
     PCollectionTuple writeRecordsResult =
         input.apply(
             "Write Records",
-            ParDo.of(new WriteRecordsDoFn(operationName, streamIdleTime, splitSize, maxRequestSize))
+            ParDo.of(
```

Comment:
I changed it to use a custom BigQueryServices implementation that has crashing behavior, PTAL. This did require changing some classes/methods to public though so that they could be accessed from the `it/gcp` module.